mother/src/test_pwm.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Test file for checking pwm pins
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/kernel.h>
+
+#include <kyvernitis.h>
+
+
+struct pwm_motor roboclaw[20] = {
+	{
+		.dev_spec = PWM_DT_SPEC_GET(DT_NODELABEL(pwm_motor_1)),
+		.min_pulse = DT_PROP(DT_NODELABEL(pwm_motor_1), min_pulse),
+		.max_pulse = DT_PROP(DT_NODELABEL(pwm_motor_1), max_pulse)
+	},
+	{
+		.dev_spec = PWM_DT_SPEC_GET(DT_NODELABEL(pwm_motor_2)),
+		.min_pulse = DT_PROP(DT_NODELABEL(pwm_motor_2), min_pulse),
+		.max_pulse = DT_PROP(DT_NODELABEL(pwm_motor_2), max_pulse)
+	},
```

Comment:
There is a better way than writing this long initialization. Can you try doing this? Its too late and I don't have the mental energy to think. 

[Zephyr docs](https://docs.zephyrproject.org/latest/build/dts/api/api.html#for-each-macros)
You can also see the dts and main.c by browsing kyvernitis from this commit: 93bb6df. It should give you a good idea.

Reply:
https://github.com/waseemR02/kyvernitis/blob/4fcba6e07b3c37dd065fc8e130f82f3a29490092/mother/src/test_pwm.c#L13-L17
Looks cleaner